### DISCUSSION TITLE
How to use dynamic select field options

### DISCUSSION THREAD

**Discussion by elmage** [payloadcms/payload#1109] [Q&A]
I was wondering if I can dynamically populate the options for a select field.
In my case, I need to get the list from outside of payload using `fetch`.

My question is, can this be achieved using hooks this way?

```
fields: [
    {
      name: 'name',
      type: 'select',
      options: [],
      hooks: {
        beforeValidate: [
          (args) => {
            // Logic to populate filed options
          },
        ],
      },
    },
  ],
```

If yes, how?

**Comment by unonweb** [comment]
Did you find any solution for this in the meantime?
I need to show different options depeding on user input (siblingData).

**Comment by DanRibbens** [comment]
There isn't a great way to do this currently, but I'll lay out a few available options and it would be interseting to get feedback on how others would want to expand on the current fields to accomodate uses like this. `beforeValidate` isn't going to work because what you're trying to do is fetch options for the frontend to show and that hook is only run server side.

1. Don't use `select`, use `text` with a custom admin.components.Field which will be more or less a copy of the `select` Input, but with your own logic on how it should fetch the available options.
2. Use a `relationship` field with a `relationTo` to the collection that you manage all the options for, this is nice because you can use the `filterOptions` to get a really nice UX without a ton of work.
3. If your dynamic options are not changing often, but are coming from an API, you could fetch them server side on start and use the retrieved data for your `options` and keep the `select`.

The select is not very dynamic, is because it uses `enum` type for the database and GraphQL schema. With that, you can't send it whatever you want and why `text` field is probably more accurate of a type for what you're trying to do.

 I would be interested in community feedback for improving `select` or other fields to help with this use-case.
